packages/cli/src/workflows/workflowSharing.service.ts

```diff
@@ -64,4 +66,33 @@ export class WorkflowSharingService {
 
 		return sharedWorkflows.map(({ workflowId }) => workflowId);
 	}
+
+	async getSharedWorkflowScopes(
+		workflowIds: string[],
+		user: User,
+	): Promise<Array<[string, Scope[]]>> {
+		const projectRelations = await this.projectRelationRepository.find({
+			where: {
+				userId: user.id,
+			},
+		});
+		const sharedWorkflows = await this.sharedWorkflowRepository.find({
+			where: {
+				workflowId: In(workflowIds),
+				projectId: In(projectRelations.map((p) => p.projectId)),
```

Comment:
We're trying to corral typeorm to the persistence layer - this wasn't flagged here because of the lint exception at L3. Can this be a method in the repository?